src/shared/util.js

```diff
@@ -644,6 +644,53 @@ function isEvalSupported() {
   }
 }
 
+/**
+ * Get the value of an inheritable property.
+ *
+ * If the PDF specification explicitly lists a property in a dictionary as
+ * inheritable, then the value of the property may be present in the dictionary
+ * itself or in one or more parents of the dictionary.
+ *
+ * If the key is not found in the tree, `undefined` is returned. Otherwise,
+ * the value for the key is returned or, if `stopWhenFound` is `false`, a list
+ * of values is returned. To avoid infinite loops, the traversal is stopped when
+ * the loop limit is reached.
+ *
+ * @param {Dict} dict - Dictionary from where to start the traversal.
+ * @param {string} key - The key of the property to find the value for.
+ * @param {boolean} getArray - Whether or not the value should be fetched as an
+ *   array. The default value is `false`.
+ * @param {boolean} stopWhenFound - Whether or not to stop the traversal when
+ *   the key is found. If set to `false`, we always walk up the entire parent
+ *   chain, for example to be able to find `\Resources` placed on multiple
+ *   levels of the tree. The default value is `true`.
+ */
+function getInheritableProperty(dict, key, getArray = false,
+                                stopWhenFound = true) {
```

Comment:
Since there's cases where you might want to use `stopWhenFound` without `getArray`, wouldn't it perhaps be a bit nicer to have this helper function take an object instead!?

```javascript
function getInheritableProperty({ dict, key, getArray = false,
                                  stopWhenFound = true, }) {
```

Reply:
Good point! I'll try to make the changes during the weekend and also re-check your pull request for version 2.0.

Reply:
> also re-check your pull request for version 2.0.

Thanks; being this close I just want to make sure that it lands *before* anything else at this point, to reduce additional risk in the release.

Reply:
Done in the new commit.